core/commands/ls.go

```diff
@@ -57,6 +65,8 @@ The JSON output contains type information.
 	Options: []cmdkit.Option{
 		cmdkit.BoolOption("headers", "v", "Print table headers (Hash, Size, Name)."),
 		cmdkit.BoolOption("resolve-type", "Resolve linked objects to find out their types.").WithDefault(true),
+		cmdkit.BoolOption(quietOptionName, "q", "Write only names."),
+		cmdkit.StringOption("type", "Type of link resolver (always|local|never)."),
```

Comment:
This should have a `WithDefault`.  From what I can tell this should be (for now) WithDefault(resolveAlways).

Reply:
This help text also feels a lot more confusing than the text for `resolve-type`. What about something like:

> Include the type of each linked object (always|local|never). If `local`, types are only determined for hashes that are pinned or in your local cache.

We could also add a note to the end of that indicating that setting it to always can be slow, so people know *why* they’d want to set this.